client/internal/receiver.go

```diff
@@ -168,3 +174,9 @@ func (r *Receiver) processErrorResponse(body *protobufs.ServerErrorResponse) {
 func (r *Receiver) rcvAddonsAvailable(addons *protobufs.AddonsAvailable) {
 	// TODO: implement this.
 }
+
+func (r *Receiver) rcvCommand(command *protobufs.ServerToAgentCommand) {
+	if command != nil {
```

Comment:
It's a nit, but `command` is essentially being checked for the second time here, i.e. this function would never be called with `nil` command

Reply:
I agree that it is redundant, but I left it that way because the first check is making it clear that if there is a command that nothing else is done in processReceivedMessage. The second check is just a parameter guard following the pattern of the other functions that do nothing if called with a nil component of the message.